src/main/java/ch/uzh/ifi/hase/soprafs23/controller/UserController.java

```diff
@@ -292,4 +296,71 @@ public void updateUserIngredients(@PathVariable Long userId,
         }
     }
 
+    @PutMapping("/users/{userId}/{groupId}/ready")
+    @ResponseStatus(HttpStatus.NO_CONTENT) // 204
+    public ResponseEntity<Void> setReady(@PathVariable Long userId, @PathVariable Long groupId, HttpServletRequest request) {
+        User user = userService.getUserById(userId);
+        Group group = groupService.getGroupById(groupId);
+
+        // 404 - user or group not found
+        if (user == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("User with id %s does not exist", userId));
+        }
+        if (group == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("Group with id %s does not exist", groupId));
+        }
+
+        // 401 - not authorized if not the same user
+        Long tokenId = userService.getUseridByToken(request.getHeader("X-Token"));
+        if(!tokenId.equals(userId)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "You are not authorized to update this user's status.");
+        }
+
+        // 401 - not authorized if not a member of the group
+        List<Long> memberIds = groupService.getAllMemberIdsOfGroup(group);
+        if(!memberIds.contains(tokenId)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, String.format("You are not a member of the group with id %d.", groupId));
+        }
+
+        // Checking the group state before setting ready state
+        GroupState groupState = group.getGroupState();
+        if(!(groupState.equals(GroupState.GROUPFORMING) || groupState.equals(GroupState.FINAL) || groupState.equals(GroupState.RECIPE))) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "The groupState must be either GROUPFORMING, FINAL or RECIPE to set ready state"); // 409
+        }
+
+        // Set user's isReady to true and save
+        user.setReady(true);
+        userRepository.save(user);
```

Comment:
please write a method for this in UserService. UserController should not need UserRepository